cmsplugin_carousel_ai/models.py

```diff
@@ -0,0 +1,91 @@
+# -*- coding: utf-8 -*-
+from django.db import models
+from django.utils.translation import ugettext_lazy as _
+
+from cms.models import CMSPlugin, Page, force_text
+from easy_thumbnails.files import get_thumbnailer
+from filer.fields.image import FilerImageField
+
+
+class Carousel(CMSPlugin):
+    """
+    Model for storing carousel meta data and configuration.
+    """
+    name = models.CharField(
+        verbose_name=_("name"),
+        max_length=160
+    )
+    interval = models.IntegerField(
+        verbose_name=_("slide changing time"),
+        default=5000,
+        help_text=_("Changing time for slides in milliseconds. 1000 ms = 1 s.",)
+    )
+
+    def __str__(self):
+        return "%s" % self.name
+
+    def copy_relations(self, oldinstance):
+        """
+        Make a copy of the related slides for the new carousel object.
+        See docs: http://docs.django-cms.org/en/release-3.3.x/how_to/custom_plugins.html#for-foreign-key-relations-from-other-objects
+        """
+        for slide in oldinstance.slides.all():
+            slide.pk = None
+            slide.carousel = self
+            slide.save()
+
+
+class Slide(models.Model):
+    """
+    Model for storing carousel slide related data.
+    """
+    carousel = models.ForeignKey(
+        Carousel,
+        verbose_name=_("carousel"),
+        related_name="slides"
+    )
+    image = FilerImageField(
+        verbose_name=_("slide image")
+    )
+    caption = models.CharField(
+        verbose_name=_("slide caption"),
+        max_length=160,
+        blank=True
+    )
+    url = models.URLField(
+        verbose_name="link to URL",
+        max_length=250,
+        blank=True
+    )
+    linked_page = models.ForeignKey(
+        Page,
+        verbose_name=_("link to page"),
+        null=True,
+        blank=True,
+        help_text=_("Page link overrides given URL."),
+        limit_choices_to={"publisher_is_draft": False},
+        related_name="linking_slides",
```

Comment:
I'm not sure there's much point in this related_name? Or rather, what's the UC for finding pages that are linked to in slides? :)